### PR TITLE
add AccessRightLevelsField, ListOfGroupsField and ListOfUsersField

### PR DESCRIPTION
**Pull request by ebeutler**
https://synjira.atlassian.net/browse/CELDEV-379

### CODE REVIEW

**Review by fpichler — CHANGES_REQUESTED** [review]
can you please add at least one TestClass using this fields in a unit test?

Edit Marc: Add a test class for each new field type.

File: src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.celements.model.classes.fields.list;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+import com.xpn.xwiki.objects.classes.LevelsClass;
+import com.xpn.xwiki.objects.classes.ListClass;
+
+@Immutable
+public final class AccessRightLevelsField extends StringListField {
```
Comment (msladek) [inline src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java]: Extend _EnumListField\<EAccessLevel>_ and overwrite _getListClass()_ and if needed also _serialize(List\<EAccessLevel>)_ and _resolveList(List\<?>)_

File: src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package com.celements.model.classes.fields.list;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+import com.xpn.xwiki.objects.classes.GroupsClass;
+import com.xpn.xwiki.objects.classes.ListClass;
+
+@Immutable
+public final class ListOfUsersField extends StringListField {
```
Comment (msladek) [inline src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java]: Extend _ListField\<XWikiUser>_

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      ret.add(EAccessLevel.getAccessLevel(elem.toString()));
+    }
+    return Collections.unmodifiableList(ret);
+  }
+
+  @Override
+  protected LevelsClass getListClass() {
+    LevelsClass element = new LevelsClass();
+    element.fromString(Joiner.on(getSeparator()).join(FluentIterable.from(Arrays.asList(
+        enumType.getEnumConstants())).transform(GET_ACCESS_LEVEL_ENUM_IDENTIFIERS)));
```
Comment (msladek) [inline src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java]: element.fromString only returns a BaseProperty, it doesn't change the LevelsClass. i guess the method should only returns a new LevelsClass?

File: src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      ret = sb.toString();
+    }
+    return ret;
+  }
+
+  @Override
+  protected List<XWikiUser> resolveList(List<?> list) {
+    List<XWikiUser> ret = new ArrayList<>();
+    for (Object elem : (Collection<?>) list) {
+      ret.add(new XWikiUser(elem.toString()));
```
Comment (msladek) [inline src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java:74]: is it a good idea at this place to check the existence of the XWikiUser doc/obj before adding it to the list?

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java (hunk 3 above)
Comment (ebeutler) [reply src/main/java/com/celements/model/classes/fields/list/AccessRightLevelsField.java]: you're right, seems not to change anything -.-

File: src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java (hunk 4 above)
Comment (ebeutler) [reply src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java:74]: @fpichler WDYT?

Comment (ebeutler) [reply src/main/java/com/celements/model/classes/fields/list/ListOfUsersField.java:74]: as discussed: not really useful at this point

**Review by fpichler — APPROVED** [review]

**Review by msladek — APPROVED** [review]

**Pull request merged**